src/main/java/org/jenkinsci/lib/configprovider/ConfigProvider.java

```diff
@@ -111,9 +112,43 @@ of this software and associated documentation files (the "Software"), to deal
     /**
      * Returns a new {@link Config} object with a unique id, starting with the id of this provider - separated by '.'. e.g. "MyCustomProvider.123456". This object is also used initialize the user
      * interface.
-     * 
+     *
      * @return the new config object, ready for editing.
+     * @deprecated use {@link #newConfig(String)}
      */
+    @Deprecated
```

Comment:
I am afraid the dance for modifying arguments of an `abstract` method is more complicated than this. You need to make both overloads concrete, one delegating to the other unconditionally, the other conditionally based on `Util.isOverridden`.